src/dc_receive_imf.rs

```diff
@@ -5116,4 +5121,61 @@ Reply from different address
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_no_private_reply_to_blocked_account() -> Result<()> {
+        let mut tcm = TestContextManager::new().await;
+        let alice = tcm.alice().await;
+        let bob = tcm.bob().await;
+
+        // =============== Bob creates a group ===============
+        let group_id =
+            chat::create_group_chat(&bob, ProtectionStatus::Unprotected, "Group").await?;
+        chat::add_to_chat_contacts_table(
+            &bob,
+            group_id,
+            bob.add_or_lookup_contact(&alice).await.id,
+        )
+        .await?;
+
+        // =============== Bob sends the first message to the group ===============
+        let sent = bob.send_text(group_id, "Hello all!").await;
+        alice.recv_msg(&sent).await;
+
+        let chats = Chatlist::try_load(&bob, 0, None, None).await.unwrap();
+        assert_eq!(chats.len(), 1);
+
+        // =============== Bob blocks Alice ================
+        //group_id.block(&bob).await?;
+        //let contact = Contact::load_from_db(context, from_id).await?;
+        Contact::block(&bob, bob.add_or_lookup_contact(&alice).await.id).await?;
+
+        // =============== Alice replies private to Bob ==============
+        let received = alice.get_last_msg().await;
+        alice.add_or_lookup_contact(&bob).await;
```

Comment:
Is this line needed? What is the point of Alice adding Bob at this point? Is it a way to unblock the chat?

Strange that it is placed between receiving a message and checking its contents.